src/main/java/io/anserini/collection/CommonCrawl100Collection.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.collection;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.*;
+import org.tukaani.xz.XZInputStream;
+
+/**
+ * A collection of files from CC-100 corpus (https://data.statmt.org/cc-100/).
+ * This can be used to read the CommonCrawl files
+ */
+
+public class CommonCrawl100Collection extends DocumentCollection<CommonCrawl100Collection.Document> {
+  private static final Logger LOG = LogManager.getLogger(CommonCrawl100Collection.class);
+  
+  public CommonCrawl100Collection(Path path){
+    this.path = path;
+    this.allowedFileSuffix = new HashSet<>(Arrays.asList(".txt.xz", ".txt"));
+  }
+  
+  @SuppressWarnings("unchecked")
+  @Override
+  public FileSegment<CommonCrawl100Collection.Document> createFileSegment(Path p) throws IOException {
+    return new Segment(p);
+  }
+  
+  /**
+   * A file in a Common Crawl collection, typically containing multiple documents.
+   */
+  public static class Segment<T extends Document> extends FileSegment<T>{
+    private JsonNode node = null;
+    private List<JsonNode> jsonNodeArray = null;
+    private Iterator<JsonNode> iterator; // iterator for JSON line objects
+  
+    public Segment(Path path) throws IOException {
+      super(path);
+  
+      if (path.toString().endsWith(".xz")) {
+        bufferedReader = new BufferedReader(new InputStreamReader(
+            new XZInputStream(new FileInputStream(path.toString()))));
+      }else{
+        bufferedReader = new BufferedReader(new InputStreamReader(
+            new FileInputStream(path.toString())));
+      }
+      
+      List<JsonNode> jsonNodeArray = getTxtFiles(bufferedReader);
+  
+      iterator = jsonNodeArray.iterator();
+      if (iterator.hasNext()) {
+        node = iterator.next();
+      }
+      
+    }
+  
+    private List<JsonNode> getTxtFiles(BufferedReader reader) throws IOException {
+      List<JsonNode> jsonNodeArray = new ArrayList<>();
+      ObjectMapper objectMapper = new ObjectMapper();
+    
+      String line;
+      int i=0;
+      while ((line = reader.readLine()) != null) {
+        try{
+          String json = "{ \"id\" : \"doc_"+i+"\", \"contents\" : \""+line.replaceAll("[-+\"\'^[\\\\p{C}]\\\\]*","").strip()+"\" }";
+          JsonNode jsonNode = objectMapper.readTree(json);
+          jsonNodeArray.add(jsonNode);
+          i++;
+        }catch (IOException e) {
```

Comment:
> hi @ToluClassics can you explain the rationale for generating a "fake" JSON document internally? Seems a bit janky... we should document the reason why this approach is justified?

Since it's a monolingual corpus; i'm just using the line ids as the document id;
doing same for the Afriberta corpus collection